libs/shared/ui/src/lib/components/inputs/input-text/input-text.tsx

```diff
@@ -45,25 +46,34 @@ export function InputText(props: InputTextProps) {
       className={className}
       onClick={() => (isInputDate ? displayPicker() : inputRef.current?.querySelector('input')?.focus())}
     >
-      <div aria-label="input-container" className={`input ${inputActions} ${isDisabled} ${hasError}`} ref={inputRef}>
-        <label htmlFor={label} className={`${hasFocus ? 'text-xs' : 'text-sm translate-y-2'}`}>
-          {label}
-        </label>
-        <input
-          data-testid={dataTestId || 'input-text'}
-          name={name}
-          id={label}
-          className="input__value"
-          defaultValue={value}
-          type={type}
-          onChange={onChange}
-          disabled={disabled}
-          onFocus={() => setFocused(true)}
-          onBlur={() => setFocused(false)}
-        />
-        {isInputDate && (
-          <div className="absolute top-1/2 -translate-y-1/2 right-4">
-            <Icon name="icon-solid-angle-down" className="text-sm text-text-500" />
+      <div className="relative">
+        <div aria-label="input-container" className={`input ${inputActions} ${isDisabled} ${hasError}`} ref={inputRef}>
+          <div className={`${disabled ? 'pointer-events-none' : ''}`}>
+            <label htmlFor={label} className={`${hasFocus ? 'text-xs' : 'text-sm translate-y-2'}`}>
+              {label}
+            </label>
+            <input
+              data-testid={dataTestId || 'input-text'}
+              name={name}
+              id={label}
+              className="input__value"
+              defaultValue={value}
+              type={type}
+              onChange={onChange}
+              disabled={disabled}
+              onFocus={() => setFocused(true)}
+              onBlur={() => setFocused(false)}
+            />
+            {isInputDate && (
+              <div className="absolute top-1/2 -translate-y-1/2 right-4">
+                <Icon name="icon-solid-angle-down" className="text-sm text-text-500" />
+              </div>
+            )}
+          </div>
+        </div>
+        {!isInputDate && rightFloatingComponent && (
+          <div data-testid="right-floating-component" className="absolute top-1/2 -translate-y-1/2 right-4">
```

Comment:
Why do you add a container with relative and not add it in the same level of isInputDate ? It should be the same thing

Reply:
It's because we are supposed to be able to interact with the float element, if it's an icon with a tooltip for example. The problem is that on disabled, we add a pointer-event-none on the whole container which make the interaction with the floating component impossible. You can't hover the tooltip, you can't click on the element, you can't do nothing. This is why I created a wrapper around that won't have the pointer-event-none on it and allow us to place the floating element.